tests/test_list_tables.py

```diff
@@ -0,0 +1,24 @@
+from . import client, connection, token, schema_main, ParentPart
+from flask.wrappers import Response
+import datajoint as dj
+
+def test_list_tables(token, client, ParentPart):
+    ScanData, ProcessScanData = ParentPart
+    REST_tables = client.post(
+        '/list_tables',
+        headers=dict(Authorization=f'Bearer {token}'),
+        json=dict(schemaName=ScanData.database)).json['tableTypeAndNames']
+    assert ScanData.__name__ == REST_tables['manual_tables'][0]
+    assert ProcessScanData.__name__ == REST_tables['computed_tables'][0]
+    assert f"""{ProcessScanData.__name__}.{
+        ProcessScanData.ProcessScanDataPart.__name__}""" == REST_tables['part_tables'][0]
+
+def test_invalid_schema_list_table(token, client):
+    # Test invalid schema
+    response: Response = client.post(
+        '/list_tables', 
+        headers=dict(Authorization=f'Bearer {token}'),
+        json=dict(schemaName='invalid_schema')
+        )
+
+    assert(response.get_data(as_text=True) in 'Database `invalid_schema` has not yet been declared. Set argument create_schema=True to create it.')
```

Comment:
Perhaps I should have been more clear about why I made my previous suggestion.

What we are doing is simply re-raising or passing through a DataJoint exception under this test case. Consider us trying to optimize this test to prevent failures when our dependencies update (DataJoint is unique since we have full control but let's consider if we had not for best practice since it will be the case with most dependencies).

Therefore, a minor update to the error message on DataJoint side would cause this test to break when all we really need is to confirm that `invalid_schema` was not created. My suggestion was to check for the least amount of words to convey the message as opposed to checking for it verbatim. We could get fancy with regex to check for only what is needed but a portion of the first sentence could be sufficient.

Actually, an even better check would be to run `list_schemas` independently and verify `invalid_schemas` is not there. You can reuse the `connection` fixture (see below). Also, if we expect this to return a non-`200` status code, we should assert this as well.

```suggestion
    assert('invalid_schema' not in dj.list_schemas(connection=connection))
```

Reply:
I am thinking it along the line of what the front end api expects, for now we are displaying that message directly as the error, thus shouldn't w check the raw string that is return in the test case instead of jumping thorugh the hoops to access list_schemas, even if the front end does use list_schemas at some point?

Reply:
@Synicix The frontend though is not expecting this message exactly, right? It is just displaying what ever it is. So we don't have anywhere that requires the exact string (good thing). Error messages tend to frequently change in packages. Which means, that we should avoid checking/parsing the string message as it is prone to failures if the wording is modified. When it has to be done, an error code is usually best practice. In any case, frontend for now shouldn't generate a case where this would happen anyway since it only makes requests based on list_schemas. This test case seems to need to confirm that a schema wasn't accidentally created. DJ's list_schemas is the right utility for this type of check.

Reply:
Fair enough point. I cahnge ti to check via list schemas and check the status code to not be 200